lib/luadata.c

```diff
@@ -50,41 +50,41 @@ static inline luadata_t *luadata_checkdata(lua_State *L, lua_Integer *offset, lu
 	return data;
 }
 
-static int luadata_getnumber(lua_State *L)
-{
-	lua_Integer offset;
-	luadata_t *data = luadata_checkdata(L, &offset, LUADATA_NUMBER_SZ);
-
-	lua_pushinteger(L, *(lua_Integer *)(data->ptr + offset));
-	return 1;
+#define LUADATA_NEWINT_GETTER(T) 	\
+static int luadata_get##T(lua_State *L) \
+{					\
+	lua_Integer offset;		\
+	luadata_t *data = luadata_checkdata(L, &offset, sizeof(T##_t));	\
```

Comment:
```
#define luadata_checkint(L, offset, T)  	luadata_checkdata((L), &(offset), sizeof(T##_t))
```

just to avoid repeat this on getter and setter..